chat.js

```diff
@@ -964,7 +964,8 @@ Chat.loadPlugins = function () {
 		let outputs = [];
 		let newMessage = '';
 		for (let i = 0; i < chatfilters.length; i++) {
-			const output = chatfilters[i].call(this, message, user, room, connection, targetUser);
+			const prevOutput = outputs[outputs.length - 1];
+			const output = chatfilters[i].call(this, message, user, room, connection, targetUser, prevOutput);
```

Comment:
Like, why isn't this just

```
message = chatfilters[i].call(this, message, user, room, connection, targetUser);
```

Reply:
Because, if one chat filter does not affect a message, it will return ``undefined``

Reply:
I mean, you could do a workaround with temporary variables to hold the original message, but I honestly think this approach is more readable.

Reply:
Something something

>Also, the output of each previous filter step should be passed to the next as the main argument.
The original message should be passed as well, as another argument.

Reply:
Yeah, you did it backwards. You passed the output of the previous filter step as another argument, rather than as the main argument.

Reply:
It's _much_ simpler to do properly, instead of the way you did it.